src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -113,11 +117,17 @@ RadixContainer<T> materialize_input(const std::shared_ptr<const Table>& in_table
   jobs.reserve(in_table->chunk_count());
 
   for (ChunkID chunk_id{0}; chunk_id < in_table->chunk_count(); ++chunk_id) {
-    jobs.emplace_back(std::make_shared<JobTask>([&, chunk_id]() {
+    const auto& chunk_guard = in_table->get_chunk(chunk_id);
```

Comment:
?

Reply:
> Unfortunately, I found some more points. This looks really hard to get right. Luckily, a failure to do so is quite noticeable when we dereference a nullptr

Ok. I'll look into this later. For now, time is up 😅

Reply:
Creating jobs for null-pointer-chunks does not make sense in my opinion.

Reply:
chunk_guard is a strange name and why not pass it to the job? And then again, why the same check in line 125?

Reply:
Addressed. Now it is `chunk_in`